### ISSUE TITLE
Loss increases with version 0.3.11

### ISSUE THREAD

**Issue by gulnazaki** [microsoft/DeepSpeed#726]
Hello,

I resumed training from a checkpoint (previously with 0.3.10) and I noticed an increase in training (and validation) loss after a few steps. (changing only the deepspeed version)

I started also training from scratch and noticed that convergence is happenning also way slower.

I installed the previous version again and everything seems to be back to normal.

No idea why this happened, I just wanted to inform you.

**Comment by tjruwase** [comment]
@gulnazaki Thanks for sharing this information. This is quite concerning. Do you mind providing more details for us to repro?

**Comment by gulnazaki** [comment]
I noticed this with a custom dataset/script, but there is a ready example that uses the same model/basic structure I can share https://github.com/lucidrains/performer-pytorch/blob/main/examples/enwik8_deepspeed/train.py

Perhaps you can reproduce it here.

I noticed it both in colab and a local (single) gpu.

**Comment by jeffra** [comment]
@gulnazaki, I'm looking over the PRs that have been merged since v0.3.10 and none of them (so far) are popping out as being extra risky related to convergence here. The only exception are 2 PRs related to our transformer kernels, are you using them in your training?

**Comment by gulnazaki** [comment]
@jeffra no

**Comment by jeffra** [comment]
Can you confirm you see the same behavior with the linked code? I just downloaded it and trying to run it now to try and repro.

**Comment by gulnazaki** [comment]
I haven't tested that with the linked code, but it should be reproduced there. 

If you want I can provide a colab with a gdrive link to the dataset/checkpoint.

**Comment by jeffra** [comment]
Hi @gulnazaki, I just tried to repro with the attached code and I don't think I am triggering the same issue. Here's the tensorboard of the two losses for v0.3.10 and v0.3.11. This is running with torch 1.6, cuda 10.2, and using a single v100-32GB GPU.

![image](https://user-images.githubusercontent.com/645595/106790362-4e0a4700-6608-11eb-8f54-d711740e1b2b.png)

**Comment by gulnazaki** [comment]
@jeffra I prepared a colab notebook you can use to reproduce it
https://colab.research.google.com/drive/1Usn5G27zTeFKEaAokADNzcOydeBCFLRg

**Comment by gulnazaki** [comment]
Here is a tensorboard for context (train and val loss) until the checkpoint.
Loss would get as high as 6.5 when continuing with 0.3.11, while now it stays around 3.5.

I tried with different lr but it would only slow down this effect.

![image](https://user-images.githubusercontent.com/38190410/106791623-d6670700-665d-11eb-89cc-3f883221b80d.png)

**Comment by jeffra** [comment]
First I tried changing the previously linked code to checkpoint and restore and that all seems fine across versions as well. The code did not support checkpointing so I did have to make changes there. I also noticed that in that example the validation data is randomly selected (https://github.com/lucidrains/performer-pytorch/blob/main/examples/enwik8_deepspeed/train.py#L113) so the restored validation loss i see is a bit different across both a restore with v0.3.10 and v0.3.11. The loss however looks fine to me between versions.

![image](https://user-images.githubusercontent.com/645595/106818487-34c7c180-662d-11eb-85de-b8753100fbda.png)

I was then able to start running the code your provided in your colab, thank you for that. It seems to show the difference immediately and is pretty dramatic right from the start. Let me dig into this further and will keep you posted.

![image](https://user-images.githubusercontent.com/645595/106821704-81fa6200-6632-11eb-9c5a-fd41281445bf.png)

**Comment by jeffra** [comment]
After a bit of digging it appears the change was introduced in #670. For various reasons prior versions of deepspeed inherited the behavior of FusedAdam from apex where by default FusedAdam uses AdamW logic. #670 updates this to make it explicit to the user which version they are using Adam or AdamW. I was able to get your model to run correctly by changing your ds_config.json to set optimizer as “AdamW”. We apologize as this change isn’t documented currently. We’re figuring out internally what to do going forward. Will update this thread after we decide what to do.

**Comment by gulnazaki** [comment]
Thanks for getting to the bottom of this, because it took me a while to figure out there was something with the new version. 

Now this makes sense.

**Comment by stas00** [comment]
https://github.com/microsoft/DeepSpeed/commit/e51e4d7/ leads to a segfaults discussed here: 
https://github.com/huggingface/transformers/issues/9996#issuecomment-773725303
its parent is fine.

**Comment by loadams** [comment]
Closing this as it appears to be a stale issue.  If you are still having issues with the latest DeepSpeed, please create a new issue and link this one and we would gladly take a look.
